backend/src/main/java/pt/ulisboa/tecnico/socialsoftware/tutor/discussion/DiscussionService.java

```diff
@@ -0,0 +1,273 @@
+            if(qAnswer.getQuizQuestion().getQuestion().getId() == questionId) {
+                return true;
+            }
+            else{
+                return false;
+            }
+        }).collect(Collectors.toList());
+
+        if(questionAnswerList.size() != 1){
+            throw new TutorException(MORE_THAN_ONE_QUESTION_ANSWER,questionId);
+        }
+
+        QuestionAnswer questionAnswer = questionAnswerList.get(0);
+        checkExistingDiscussion(questionAnswer);
+
+        Discussion discussion = new Discussion(user, questionAnswer, discussionDto);
+        this.entityManager.persist(discussion);
+        List<ReplyDto> replies = discussionDto.getReplies();
+        if (replies != null && !replies.isEmpty()) {
+            for (ReplyDto reply : replies) {
+                User student = userRepository.findById(reply.getUserId())
+                        .orElseThrow(() -> new TutorException(USER_NOT_FOUND, reply.getUserId()));
+
+                this.entityManager.persist(new Reply(student, reply, discussion));
+            }
+        }
+
+        return new DiscussionDto(discussion);
+    }
+
+    private void checkExistingDiscussion(QuestionAnswer questionAnswer) {
+        if (questionAnswer.getDiscussion() != null) {
+            throw new TutorException(DUPLICATE_DISCUSSION);
+        }
+    }
+
+    @Retryable(value = { SQLException.class }, backoff = @Backoff(delay = 5000))
+    @Transactional(isolation = Isolation.REPEATABLE_READ)
+    public DiscussionDto changeStatus(int discussionId) {
+        Discussion discussion = discussionRepository
+                .findById(discussionId)
+                .orElseThrow(() -> new TutorException(DISCUSSION_NOT_FOUND, discussionId));
+
+        if((discussion.getReplies().isEmpty() || discussion.getReplies() == null) && !discussion.isClosed()){
+            throw new TutorException(CLOSE_NOT_POSSIBLE);
+        }
+        discussion.changeStatus();
+        return new DiscussionDto(discussion);
+    }
+
+    @Retryable(value = { SQLException.class }, backoff = @Backoff(delay = 5000))
+    @Transactional(isolation = Isolation.REPEATABLE_READ)
+    public DiscussionDto changeReplyAvailability(int replyId) {
+        Reply reply = replyRepository
+                .findById(replyId)
+                .orElseThrow(() -> new TutorException(REPLY_NOT_FOUND, replyId));
+
+        reply.changeAvailability();
+        return new DiscussionDto(reply.getDiscussion());
+    }
+
+    @Retryable(value = { SQLException.class }, backoff = @Backoff(delay = 5000))
+    @Transactional(isolation = Isolation.REPEATABLE_READ)
+    public ReplyDto createReply(DiscussionDto discussionDto, ReplyDto replyDto) {
+        checkDiscussionDto(discussionDto);
+
+        User user = userRepository.findById(replyDto.getUserId()).orElseThrow(()->new TutorException(USER_NOT_FOUND, replyDto.getUserId()));
+
+        checkUserAndDiscussion(user, discussionDto);
```

Comment:
but the user does not need to be the user of the student... check inside the reply constructor